### PR TITLE
Refactors timeline component

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/10155

- separates timeline into two separate components: phases and milestone
- queries now live in `ProjectTimeline` and `DashboardTimelineModal` and data is passed down into child components (phases and milestones)

## Testing
**URL to test:** 
https://deploy-preview-776--atd-moped-main.netlify.app/

**Steps to test:**
1. go to dashboard, click the phase badge, add or edit phase, badge on dashboard should change accordingly
2. repeat step 1 for milestones meter
3. go to a project, click timeline tab, add or edit phase, badge on summary tab should change accordingly
4. repeat step 2 with milestones

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js [hunk 1]
```diff
@@ -26,15 +33,27 @@ const DashboardTimelineModal = ({
...
+   * @type {boolean} - loading state
+   * @type {object} - details and messages when there is a query error
+   * @type {object} - data returned from Hasura
+   * @function refetch - Provides a manual callback to update the Apollo cache
+   * */
+  const { loading, error, data, refetch } = useQuery(TIMELINE_QUERY, {
+    variables: { projectId },
+    fetchPolicy: "no-cache",
+  });
+
```
Comment (mateoclarke) [inline moped-editor/src/views/dashboard/DashboardTimelineModal.js:53]: 💚 Love to see these JSDoc comments

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js [hunk 2]
```diff
@@ -58,10 +77,32 @@ const DashboardTimelineModal = ({
...
-          <ProjectTimeline
-            table={table}
-            projectIdFromTimelineModal={projectId}
-          />
+          <ApolloErrorHandler error={error}>
+            <CardContent>
+              <Grid container spacing={2}>
+                <Grid item xs={12}>
+                  <Box mb={2} style={{ maxWidth: "100%" }}>
+                    {table === "phases" && (
```
Comment (mateoclarke) [inline moped-editor/src/views/dashboard/DashboardTimelineModal.js:85]: chefs-kiss-french-chef

**Review by mateoclarke — APPROVED** [review]
Love this refactor. So much more organized and reusable! Thanks for having the patience and persistence on this. I didn't test the UI and I know you were working on a few last bug fixes, but just wanted to give positive feedback on the code. 👍

File: moped-editor/src/views/dashboard/DashboardTimelineModal.js (hunk 2 above)
Comment (tillyw) [reply moped-editor/src/views/dashboard/DashboardTimelineModal.js:85]: > Love this refactor. So much more organized and reusable! Thanks for having the patience and persistence on this. I didn't test the UI and I know you were working on a few last bug fixes, but just wanted to give positive feedback on the code. 👍

thanks @mateoclarke!! :) all the bug fixes are in the pr so it should be working smoothly! thanks for the review, hope all is going well!

File: moped-editor/src/views/projects/projectView/ProjectMilestones.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+  // If the query is loading or data object is undefined,
+  // stop here and just render the spinner.
+  if (loading || !data) return <CircularProgress />;
+
+  /**
+   * Phase table lookup object formatted into the shape that Dropdown expects
+   * Ex: { 1: "Potential", 2: "Planned", ...}
+   */
+  const phaseNameLookup = data.moped_phases.reduce(
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectMilestones.js:65]: for code like this that exists in both milestones and phase components, i would prefer to see this as a function imported from a `utils.js` file in this directory.

File: moped-editor/src/views/projects/projectView/ProjectMilestones.js [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+      type="date"
+      variant="standard"
+      value={props.value}
+      onChange={(e) => props.onChange(e.target.value)}
+      onKeyDown={(e) => handleKeyEvent(e)}
+      InputLabelProps={{
+        shrink: true,
+      }}
+    />
+  );
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectMilestones.js:104]: for **components** that are identical in both phases and milestones, would prefer to see individual component modules in this directory. `DateFieldEditComponent.js`, `ToggleEditComponent.js`, etc.

File: moped-editor/src/views/projects/projectView/ProjectPhases.js [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+  );
+
+  /**
+   * DropDownSelectComponent - Renders a drop down menu for MaterialTable
+   * @param {object} props - Values passed through Material Table `editComponent`
+   * @param {string} name - Field name
+   * @return {JSX.Element}
+   * @constructor
+   */
+  const DropDownSelectComponent = (props) => {
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectPhases.js:214]: I know this is pre-existing code, and this component is not used in Milestones, but it really should be extracted to a separate file.

**Review by johnclary — CHANGES_REQUESTED** [review]
@tillyw thank you for breaking apart this beastly component! i would like to see the redundant chunks of code split out, but defer to @chiaberry if we should table it as tech debt or not.

**Comment by tillyw** [comment]
@johnclary I refined the code as you requested, please have another look!

**Review by johnclary — APPROVED** [review]
beautiful—thank you for tightening this up!

🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**